cli/src/cli.rs

```diff
@@ -1146,43 +1033,35 @@ fn process_pay(
     };
 
     if timestamp == None && *witnesses == None {
+        let nonce_authority = nonce_authority.unwrap_or_else(|| config.keypair.as_ref());
         let mut tx = if let Some(nonce_account) = &nonce_account {
-            let nonce_authority: &Keypair = nonce_authority
-                .map(|authority| authority.keypair())
-                .unwrap_or(&config.keypair);
             system_transaction::nonced_transfer(
-                &config.keypair,
+                config.keypair.as_ref(),
                 to,
                 lamports,
                 nonce_account,
                 nonce_authority,
                 blockhash,
             )
         } else {
-            system_transaction::transfer(&config.keypair, to, lamports, blockhash)
+            system_transaction::transfer(config.keypair.as_ref(), to, lamports, blockhash)
         };
 
-        if let Some(signers) = signers {
-            replace_signatures(&mut tx, &signers)?;
```

Comment:
Looks like `fn replace_signatures` is dead code now. I'm confused why we don't need this anymore, could you please explain?